.github/workflows/build-and-publish-master.yml

```
@@ -0,0 +1,25 @@
+# On tag push, build and publish new release
+
+name: GitHub Actions - Tagged release
+
+on:
+  push:
+    branches:
+      - 'main'
+
+jobs:
+  build-and-publish-sdk:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+
+      - uses: actions/setup-java@v3
+        with:
+          distribution: 'corretto'
+          java-version: '8'
+          cache: 'gradle'
+
+      - name: 'Build'
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        run: ./gradlew build publish -Pversion="${GITHUB_REF_NAME:1}" --no-daemon
```

This workflow wont have a valid tag. Consider using tags for versioning or have a file like `VERSION[.md]` and read version from the file
```suggestion
        run: ./gradlew build publish -Pversion="$(cat VERSION)" --no-daemon
```